### PR TITLE
merge the code for the two model types

### PR DESCRIPTION
**Pull request by ThomVett**
remove duplicated code for the survival model and the classification model

### CODE REVIEW

File: webapp/service/machine_learning.py [hunk 1]
```diff
@@ -95,3 +102,66 @@ def concatenate_modalities_rois(features_df):
...
+
+    training_id = get_training_id(extraction_id, collection_id)
+
+    model = ModelClass(
+        feature_extraction_id=extraction_id,
+        collection_id=collection_id,
+        album=album,
+        feature_selection=feature_selection,
+        feature_names=features_df.columns,  # TODO - This might change with feature selection
+        estimator_step=ESTIMATOR_STEP.CLASSIFICATION.value,
```
Comment (roger-schaer) [inline webapp/service/machine_learning.py:153]: This shouldn't be hard-coded to "ESTIMATOR_STEP.CLASSIFICATION", but should be selected between ESTIMATOR_STEP.CLASSIFICATION & ESTIMATOR_STEP.SURVIVAL depending on the model type.

**Review by roger-schaer — CHANGES_REQUESTED** [review]
Nice refactoring, I think it's good to cut down on this duplicate code, I just found one issue where the estimator step was hard-coded to the classification model (see comment in machine_learning.py)

File: webapp/service/machine_learning.py (hunk 1 above)
Comment (ThomVett) [reply webapp/service/machine_learning.py:153]: ah yes good catch will change thx

Comment (ThomVett) [reply webapp/service/machine_learning.py]: done!

**Pull request merged**